src/library/itunes/itunesimporter.h

```diff
@@ -2,22 +2,23 @@
 
 #include <memory>
 
-#include "library/treeitem.h"
+class TreeItem;
+class ITunesFeature;
 
 struct ITunesImport {
     std::unique_ptr<TreeItem> playlistRoot;
 };
 
 class ITunesImporter {
   public:
+    ITunesImporter(ITunesFeature* pParentFeature);
```

Comment:
Hm, not sure how I feel about the cyclic dependency here (and in the DAO). Perhaps we could at least turn this into an interface for "cancellables" to avoid the forward declarations?

Reply:
Both inherited classes had an ITunesFeature* member anyway. Moving it into the base class is just a change for code deduplication. 

Defining an interface could be an option. I am just not the right person to do any more refactoring, because I have no access to a macOS build device.

Reply:
Does it need to be in the base class though? Since `ITunesImporter` is interface-ish, we generally don't instantiate it directly and I don't see much benefit in moving the member up there (we wouldn't really save on boilerplate and the fact that the interface is unaware of the parent feature was a more or less intentional choice to precisely avoid this cyclic dependency involving forward declarations etc)

Reply:
whats the ownership relationship like now? We should either document that or use an explicit alias such as [`observer_ptr`](https://en.cppreference.com/w/cpp/experimental/observer_ptr).